backend/src/main/java/com/bakdata/conquery/mode/local/LocalNamespaceHandler.java

```diff
@@ -1,38 +1,64 @@
 package com.bakdata.conquery.mode.local;
 
+import java.io.IOException;
+
 import com.bakdata.conquery.io.storage.MetaStorage;
 import com.bakdata.conquery.io.storage.NamespaceStorage;
 import com.bakdata.conquery.mode.InternalObjectMapperCreator;
 import com.bakdata.conquery.mode.NamespaceHandler;
 import com.bakdata.conquery.mode.NamespaceSetupData;
 import com.bakdata.conquery.models.config.ConqueryConfig;
+import com.bakdata.conquery.models.config.DatabaseConfig;
+import com.bakdata.conquery.models.config.SqlConnectorConfig;
 import com.bakdata.conquery.models.identifiable.ids.specific.DatasetId;
 import com.bakdata.conquery.models.index.IndexService;
 import com.bakdata.conquery.models.query.ExecutionManager;
 import com.bakdata.conquery.models.worker.LocalNamespace;
-import com.bakdata.conquery.sql.SqlContext;
+import com.bakdata.conquery.sql.DSLContextWrapper;
+import com.bakdata.conquery.sql.DslContextFactory;
 import com.bakdata.conquery.sql.conquery.SqlExecutionManager;
+import com.bakdata.conquery.sql.conversion.SqlConverter;
+import com.bakdata.conquery.sql.conversion.dialect.SqlDialect;
+import com.bakdata.conquery.sql.conversion.dialect.SqlDialectFactory;
+import com.bakdata.conquery.sql.execution.ResultSetProcessorFactory;
+import com.bakdata.conquery.sql.execution.SqlExecutionResult;
 import com.bakdata.conquery.sql.execution.SqlExecutionService;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jooq.DSLContext;
 
 @RequiredArgsConstructor
+@Slf4j
 public class LocalNamespaceHandler implements NamespaceHandler<LocalNamespace> {
 
 	private final ConqueryConfig config;
 	private final InternalObjectMapperCreator mapperCreator;
-	private final SqlContext sqlContext;
-	private final SqlExecutionService sqlExecutionService;
+	private final SqlDialectFactory dialectFactory;
 
 	@Override
 	public LocalNamespace createNamespace(NamespaceStorage namespaceStorage, MetaStorage metaStorage, IndexService indexService) {
+
 		NamespaceSetupData namespaceData = NamespaceHandler.createNamespaceSetup(namespaceStorage, config, mapperCreator, indexService);
-		ExecutionManager executionManager = new SqlExecutionManager(sqlContext, sqlExecutionService, metaStorage);
+		SqlConnectorConfig sqlConnectorConfig = config.getSqlConnectorConfig();
+		// TODO mapping between a namespace/dataset and a database config - how do we create this mapping?
```

Comment:
Da sind wir auf euer Feedback angewiesen: Wie wollen wir ein Mapping zwischen Dataset/Namespace und Datenbank-Verbindung herstellen?

Reply:
Ich denke, es wäre gut wenn das an dieser Stelle aus dem Storage ausgelesen wird.
Da bedeuted, dass es bei einem neu erstellten Dataset schon vorher in den Storage reingeschrieben werden muss (in DatasetRegistry#createNamespace(Dataset dataset, Validator validator)).

Ich würde hier aber dazu tendieren nicht die kompletten Db-Connection Daten (url, username, pwd) in den Storage zu persistieren, sondern nur einen `key`, mit dem man dann `sqlConnectorConfig.getDatabaseConfigs().get(key)` die Verbindung aus der Config ausließt.

Ich frage mich bei dem Ansatz noch wass wir mit dem DB-Name machen, der sich ja für jedes Dataset unterscheiden wird. Und ich möchte nicht für jedes neue Dataset, dass ich anlege, schon vorher in der Config einen Eintrag gemacht haben müssen.

Naive Idee wäre, dass die DatasetId dem Datenbanknamen entsprechen muss, aber vielleicht sollten wir auch das Dataset Objekt um die Info erweitern.